### PR TITLE
SAK-32285 Create audit database records when adding or deleting users…

### PR DESCRIPTION
**Pull request by RebeccaOxfordUni**
… via the entity broker.

### CODE REVIEW

**Comment by buckett** [comment]
Build is failing for this one and does look related.

**Comment by lardo-de-arnaud** [comment]
can somebody review this please

File: entitybroker/core-providers/src/java/org/sakaiproject/entitybroker/providers/MembershipEntityProvider.java [hunk 1]
```diff
@@ -21,18 +21,11 @@
...
-import java.util.Collection;
-import java.util.Collections;
-import java.util.Comparator;
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.Iterator;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
```
Comment (bjones86) [inline entitybroker/core-providers/src/java/org/sakaiproject/entitybroker/providers/MembershipEntityProvider.java]: Please no * imports, they can cause more problems than they're worth.

File: entitybroker/core-providers/src/java/org/sakaiproject/entitybroker/providers/MembershipEntityProvider.java [hunk 2]
```diff
@@ -79,7 +74,7 @@
 RESTful, ActionsExecutable {
 
     private static Logger log = LoggerFactory.getLogger(MembershipEntityProvider.class);
-
+
```
Comment (bjones86) [inline entitybroker/core-providers/src/java/org/sakaiproject/entitybroker/providers/MembershipEntityProvider.java]: Unnecessary whitespace change here.

File: entitybroker/core-providers/src/java/org/sakaiproject/entitybroker/providers/MembershipEntityProvider.java [hunk 3]
```diff
@@ -175,13 +181,21 @@ public boolean unjoinCurrentUserFromSite(EntityView view, Map<String, Object> pa
...
                     "siteId must be set in order to unjoin sites, set in params or in the URL /unjoin/site/siteId");
         }
         checkSiteSecurity(siteId);
         try {
             siteService.unjoin(siteId);
+            //String user = sessionManager().getCurrentSessionUserId();
+            String currentUserEid = userEntityProvider.getCurrentUser(view).getEid(); //userDirectoryService.getCurrentUser().getEid();
+            String roleId = siteService.getSite(siteId).getJoinerRole();
+            String[] userAuditString = {siteId,currentUserEid,roleId,UserAuditService.USER_AUDIT_ACTION_REMOVE,userAuditRegistration.getDatabaseSourceKey(),currentUserEid};
+            List<String[]> userAuditList = new ArrayList<>();
```
Comment (bjones86) [inline entitybroker/core-providers/src/java/org/sakaiproject/entitybroker/providers/MembershipEntityProvider.java]: These two lines can be combined; `Arrays.asList....`

Also, is there a reason you're creating a List of String arrays? Why not `List<List<String>>`?

File: entitybroker/core-providers/src/java/org/sakaiproject/entitybroker/providers/MembershipEntityProvider.java [hunk 4]
```diff
@@ -840,11 +854,16 @@ public String createEntity(EntityReference ref, Object entity, Map<String, Objec
...
 
+        String[] userAuditString;
+        List<String[]> userAuditList = new ArrayList<>();
+        
         // check for a batch add
         String[] userIds = checkForBatch(params, userId);
-        // now add all the memberships
         String memberId = "";
         String currentUserId = developerHelperService.getCurrentUserId();
+        User user = null;
```
Comment (bjones86) [inline entitybroker/core-providers/src/java/org/sakaiproject/entitybroker/providers/MembershipEntityProvider.java]: I don't think you're using this `User` object outside of the `for` loop below, so I would move it inside that loop scope.

File: entitybroker/core-providers/src/java/org/sakaiproject/entitybroker/providers/MembershipEntityProvider.java (hunk 1 above)
Comment (RebeccaOxfordUni) [inline entitybroker/core-providers/src/java/org/sakaiproject/entitybroker/providers/MembershipEntityProvider.java]: I've changed this.

File: entitybroker/core-providers/src/java/org/sakaiproject/entitybroker/providers/MembershipEntityProvider.java (hunk 2 above)
Comment (RebeccaOxfordUni) [inline entitybroker/core-providers/src/java/org/sakaiproject/entitybroker/providers/MembershipEntityProvider.java]: Dealt with this too.

File: entitybroker/core-providers/src/java/org/sakaiproject/entitybroker/providers/MembershipEntityProvider.java (hunk 3 above)
Comment (RebeccaOxfordUni) [inline entitybroker/core-providers/src/java/org/sakaiproject/entitybroker/providers/MembershipEntityProvider.java]: Using Arrays.asList requires this code which looks too awful:
Arrays.asList(new String[][]{new String[]{siteId,currentUserEid,roleId, UserAuditService.USER_AUDIT_ACTION_REMOVE,userAuditRegistration.getDatabaseSourceKey(),currentUserEid}})
so I've used a Collections method instead.

The method addToUser takes a list of string arrays and is used by other code so I haven't changed this.

File: entitybroker/core-providers/src/java/org/sakaiproject/entitybroker/providers/MembershipEntityProvider.java (hunk 4 above)
Comment (RebeccaOxfordUni) [inline entitybroker/core-providers/src/java/org/sakaiproject/entitybroker/providers/MembershipEntityProvider.java]: Change made.
